### ISSUE TITLE
TypeError: '<' not supported between instances of '_OrderedBlock' and '_OrderedBlock'

### ISSUE THREAD

**Issue by wrvsrx** [SEIAROTg/autobean-format#2]
# Reproduce

1. install `autobean-format` via `pip install git+https://github.com/SEIAROTg/autobean-format`
1.  create `main.bean`

 ```
2023-01-02 * "aa" "bb"
    Assets:B    -3 CNY
    Expenses:A

2023-01-02 * "ss" "ss"
    Assets:B    -1 CNY
    Expenses:A

2023-01-01 * "aa"
    Assets:B     5 CNY
    Income:A

2023-01-01 * "bb" "bb"
    Expenses:A  -1 CNY
    Expenses:B
 ```

1. `autobean-format --sort main.bean`
1. It will throw error

```
Traceback (most recent call last):
  File "/nix/store/928i1hcwss8iw3al9x32svhys86b60zd-python3.11-autobean-format-b2137ffd5977c4eacd64fc0c457aa5a7ffbc4b5c/bin/.autobean-format-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/main.py", line 80, in main
    formatted = formatter.format_file(file)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/main.py", line 53, in format_file
    formatter.format(file.model, self._parser, self._options, stream)
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/formatter.py", line 16, in format
    for token in formatters.format(model, context):
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/formatters/base.py", line 39, in format
    yield from formatter(model, context)
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/formatters/file.py", line 75, in format_file
    blocks = sorting.sort_blocks(blocks)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/internal/sorting.py", line 340, in sort_blocks
    results.extend(_sort_compartment(compartment))
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/internal/sorting.py", line 318, in _sort_compartment
    blocks = _OrderedBlock.sort(blocks)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/internal/sorting.py", line 110, in sort
    return list(cls.merge(sorted, unsorted))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/ydgdb3zikhsznnv4ggbba1sq4201wkza-python3-3.11.2-env/lib/python3.11/site-packages/autobean_format/internal/sorting.py", line 293, in merge
    heapq.heapify(keyed_unsorted)
TypeError: '<' not supported between instances of '_OrderedBlock' and '_OrderedBlock'
```

**Comment by SEIAROTg** [comment]
Thanks for flagging this.

This particular example has been fixed in b1222f3ca1bfc771baca276876c3c6df99911bc4. Can you verify if it works now and close this if so?

**Comment by wrvsrx** [comment]
It works now. Thanks!
